chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +932,210 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+        change: uint64 = uint64(spend_value - total_amount)
+        xch_spends = []
+        if xch_change_ph is None:
+            xch_change_ph = await self.standard_wallet.get_new_puzzlehash()
+        xch_primaries = [
+            AmountWithPuzzlehash({"puzzlehash": xch_change_ph, "amount": change, "memos": [xch_change_ph]})
+        ]
+        for xch_coin in xch_coins:
+            puzzle: Program = await self.standard_wallet.puzzle_for_puzzle_hash(xch_coin.puzzle_hash)
+            solution: Program = self.standard_wallet.make_solution(
+                primaries=xch_primaries,
+                fee=fee,
+            )
+            xch_spends.append(CoinSpend(xch_coin, puzzle, solution))
+        xch_spend = await self.standard_wallet.sign_transaction(xch_spends)
+
+        innerpuz: Program = did_wallet.did_info.current_inner
+        # Quote message puzzle & solution
+        if new_innerpuzhash is None:
+            new_innerpuzhash = innerpuz.get_tree_hash()
+
+        # create primaries
+        primaries = [
+            AmountWithPuzzlehash(
+                {"puzzlehash": new_innerpuzhash, "amount": uint64(did_coin.amount), "memos": [new_innerpuzhash]}
+            )
+        ]
+
+        n = len(metadata_list)
+        amount = uint64(1)
+        zero_coin_spends = []
+        launcher_spends = []
+        launcher_ids = []
+        eve_spends = []
+        p2_inner_puzzle = await self.standard_wallet.get_new_puzzle()
+        ending_num = starting_num + n
+        for m in range(starting_num, ending_num):
+            zero_coin_puz = did_wallet_puzzles.DID_NFT_LAUNCHER_MOD.curry(
+                did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, m, max_num
+            )
+            primaries.append(
+                AmountWithPuzzlehash(
+                    {"puzzlehash": zero_coin_puz.get_tree_hash(), "amount": uint64(0), "memos": [zero_coin_puz]}
+                )
+            )
+            zero_coin_sol = Program.to([did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, m, n])
+            zero_coin = Coin(did_coin.name(), zero_coin_puz.get_tree_hash(), uint64(0))
+            zero_coin_spend = CoinSpend(zero_coin, zero_coin_puz, zero_coin_sol)
+            zero_coin_spends.append(zero_coin_spend)
+            launcher_coin = Coin(zero_coin.name(), did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, amount)
+            launcher_ids.append(launcher_coin.name())
+            metadata = metadata_list[m - starting_num - 1]
+            inner_puzzle = create_ownership_layer_puzzle(
+                launcher_coin.name(),
+                b"",
+                p2_inner_puzzle,
+                metadata["royalty_pc"],
+                royalty_puzzle_hash=metadata["royalty_ph"],
+            )
+            eve_fullpuz = nft_puzzles.create_full_puzzle(
+                launcher_coin.name(), metadata["program"], NFT_METADATA_UPDATER.get_tree_hash(), inner_puzzle
+            )
+            announcement_set: Set[Announcement] = set()
```

Comment:
This is dangerous!
You're clearing the set inside the loop, and you aren't asserting these announcements from a signed spend.

If these announcements aren't asserted from the signed spend that creates them then anybody can change what gets created